rust/src/examples.rs

```diff
@@ -0,0 +1,214 @@
+accept. By default, they are set to the theoretical minimum in order for the
+protocol to function correctly.
+
+- To complement, the settings [`min_sum_time`] and
+[`min_update_time`] specify, respectively, the minimum amount of time (in
+seconds) the coordinator should wait for `sum`/`sum2` and `update` messages. To
+allow for more messages to be processed, increase these times.
+
+- [`expected_participants`] should be set to the expected number of participants
+for the session.
+
+The [`MaskSettings`] determines the masking configuration, consisting of the
+group type, data type, bound type and model type. The [`ModelSettings`] specify
+the size of the model used. Both of these settings should be decided in advance
+with participants, and agreed upon by both.
+
+## Running
+
+The coordinator can be run as follows:
+
+```ignore
+$ git clone git://github.com/xaynetwork/xaynet
+$ cd xaynet/rust
+$ cargo run --bin coordinator -- -c ../configs/config.toml
+```
+
+We will assume `configs/config.toml` includes the values shown in the excerpt
+above.
+
+# Connecting with Participants
+
+The below shows an example of how participants can be created to connect to the
+coordinator running as instructed above. It is a slightly adapted version of
+the `test-drive-net` example, for illustrative purposes.
+
+```no_run
+use tokio::signal;
+use xaynet::{
+    client::{Client, ClientError},
+    mask::{FromPrimitives, Model},
+};
+
+#[tokio::main]
+async fn main() -> Result<(), ClientError> {
+    let len = 4_usize;
+    let model = Model::from_primitives(vec![0; len].into_iter()).unwrap();
+
+    let mut clients = Vec::with_capacity(20_usize);
+    for id in 0..20 {
+        let mut client = Client::new_with_addr(1, id, "http://127.0.0.1:8081")?;
+        client.local_model = Some(model.clone());
+        let join_hdl = tokio::spawn(async move {
+            tokio::select! {
+                _ = signal::ctrl_c() => {}
+                result = client.start() => {
+                    println!("{:?}", result);
+                }
+            }
+        });
+        clients.push(join_hdl);
+    }
+
+    for client in clients {
+        let _ = client.await;
+    }
+
+    Ok(())
+}
+```
+
+As this example is meant solely as a demonstration of the operation of
+the PET protocol, it does not perform any training as such (see other
+forthcoming examples for that). Instead, a "dummy" zero-model is used throughout.
+Nevertheless, note that its length and contents are (respectively) required to
+match the size and mask configuration expected by the coordinator.
+
+The example creates the expected number of participants (called [`Client`]s
+here), connecting to the address where the coordinator should be listening. The
+participants are then started, and run continuously until stopped (with CTRL +
+C).
+
+## Running
+
+The actual `test-drive-net` example is a tidier version of the above, where the
+hard-coded numbers are made configurable. To run:
+
+```ignore
```

Comment:
```suggestion
```text
```
I think you can use `text` here. That cloud fix the problem with the CI

Reply:
thanks! this does work-around the smoke-test complaint too